Source/WebCore/page/linux/ResourceUsageThreadLinux.cpp

```diff
@@ -312,10 +313,15 @@ void ResourceUsageThread::platformCollectMemoryData(JSC::VM* vm, ResourceUsageDa
     data.categories[MemoryCategory::GCHeap].dirtySize = currentGCHeapCapacity;
     data.categories[MemoryCategory::GCOwned].dirtySize = currentGCOwnedExtra - currentGCOwnedExternal;
     data.categories[MemoryCategory::GCOwned].externalSize = currentGCOwnedExternal;
-    size_t categoriesTotalSize = 0;
-    for (auto& category : data.categories)
-        categoriesTotalSize += category.totalSize();
-    data.categories[MemoryCategory::Other].dirtySize = data.totalDirtySize - categoriesTotalSize;
```

Comment:
I was thinking about that and I was considering to leave it. MemoryCategory::Other will contain what is not in other categories(JS, images, page, layers).

Reply:
data.totalDirtySize is calculated base on data from /proc/self/statm (resident-share):
resident   (2) resident set size
                             (inaccurate; same as VmRSS in /proc/[pid]/status)
shared     (3) number of resident shared pages
                             (i.e., backed by a file)
                             (inaccurate; same as RssFile+RssShmem in
                             /proc/[pid]/status)

while in my change LibcMalloc category is:
data.categories[MemoryCategory::LibcMalloc].dirtySize = WTF::fastMallocStatistics().committedVMBytes;
It is taken from "getrusage" (ru_maxrss) - https://github.com/WebKit/WebKit/blob/main/Source/WTF/wtf/FastMalloc.cpp#L677
ru_maxrss;        /* maximum resident set size */ 

I have checked that in case of Win and Mac, the LibcMalloc category is calculated in different way, but IMO it could be reused also in case of Linux:
(Mac): https://github.com/WebKit/WebKit/blob/main/Source/WebCore/page/cocoa/ResourceUsageThreadCocoa.mm#L234
(Win): https://github.com/WebKit/WebKit/blob/main/Source/WebCore/page/win/ResourceUsageThreadWin.cpp#L133

@shivamidow : what do you think?

Reply:
I'm not sure I understand this discussion, but I do't understand why the other category is removed, it's supposed to be the rest of the memory not included in categories.